server/src/main/java/org/opensearch/search/deciders/ConcurrentSearchDeciders.java

```diff
@@ -0,0 +1,159 @@
+ */
+public class ConcurrentSearchDeciders {
+    private static final Logger logger = LogManager.getLogger(ConcurrentSearchDeciders.class);
+
+    private Collection<ConcurrentSearchDecision> concurrentSearchDecisions = new ArrayList<>();;
+
+    private Collection<ConcurrentSearchDecider> concurrentSearchDeciders;
+
+    private ConcurrentSearchDecisionVisitor concurrentSearchDecisionVisitor;
+
+    private SearchContext searchContext;
+    private ClusterSettings clusterSettings;
+    private IndexSettings indexSettings;
+
+    public ConcurrentSearchDeciders(
+        Collection<ConcurrentSearchDecider> concurrentSearchDeciders,
+        SearchContext searchContext,
+        ClusterSettings clusterSettings,
+        IndexSettings indexSettings
+    ) {
+        this.concurrentSearchDeciders = concurrentSearchDeciders;
+        this.concurrentSearchDecisionVisitor = new ConcurrentSearchDecisionVisitor();
+        this.searchContext = searchContext;
+        this.indexSettings = indexSettings;
+        this.clusterSettings = clusterSettings;
+    }
+
+    /**
+     * Returns the boolean result combining the results of all registered {@link ConcurrentSearchDecider} implementations
+     * using the {@link ConcurrentSearchDecision#getCompositeDecision} method
+     * <br/>
+     * Decision is collected in the following fashion -<br/>
+     * 1. Filter out deciders that opt-out of the decision-making process<br/>
+     * 2. Collect the decision from the {@link DefaultConcurrentSearchDecider}<br/>
+     * 3. Collect the decision for any other remaining deciders.<br/>
+     *
+     */
+    public boolean getConcurrentSearchDecisionForSearchRequest() {
+
+        Collection<ConcurrentSearchDecider> decidersToConsider = new ArrayList<>(concurrentSearchDeciders);
+
+        // if the search request is on an index that certain deciders don't care about,
+        // those deciders can opt out of decision-making process. We can filter out such deciders
+        concurrentSearchDeciders.forEach(concurrentSearchDecider -> {
+            if (concurrentSearchDecider.optOutOfDecisionMakingForIndex(indexSettings)) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("{} decider opt out of decision making", concurrentSearchDecider.getClass());
+                }
+                decidersToConsider.remove(concurrentSearchDecider);
+            }
+            if (concurrentSearchDecider instanceof DefaultConcurrentSearchDecider) {
+                ConcurrentSearchDecision decision = concurrentSearchDecider.getConcurrentSearchDecisionFromOperationType(
+                    searchContext,
+                    indexSettings,
+                    clusterSettings,
+                    null
+                );
+                concurrentSearchDecisions.add(decision);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("{} decider returned decision {}", concurrentSearchDecider.getClass(), decision.toString());
```

Comment:
It should always be the `DefaultConcurrentSearchDecider` here right

Reply:
Yes. I could hard code that in or maybe add an assert?